compiler/src/iree/compiler/Utils/Shape.h

```diff
@@ -0,0 +1,31 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_UTILS_SHAPE_H_
+#define IREE_COMPILER_UTILS_SHAPE_H_
+
+#include "llvm/ADT/STLExtras.h"
+#include "mlir/IR/BuiltinTypeInterfaces.h"
+
+namespace mlir::iree_compiler {
+
+// Unranked shapes are always considered to have more dynamic dimensions than
+// ranked.
+inline bool shapeHasLessOrEqualDynamicDimensions(ShapedType t1, ShapedType t2) {
```

Comment:
[YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)
on the 3rd or 4th time a 3 line block of code is used it's worth hoisting into a global shared util - but there's a bar for doing this - every shared util pollutes the project and adds a maintenance burden and has to be worth it. A single use in a single location is not worth it. Just inline this as static where this is used.

(this would also need to be static here - you can't put inline functions in header files)

Reply:
I moved it.